src/scippnexus/base.py

```
@@ -140,6 +141,30 @@ def assemble(
         """
         return dg
 
+    def _convert_index_to_positional(self, sel):
+        if isinstance(sel, dict):
+            return dict((self._convert_index_to_positional(s) for s in sel.items()))
+        if not (isinstance(sel, tuple) and len(sel) > 1):
+            return sel
+        coord, index = sel
+        if coord not in self._children:
+            return sel
```

I think we should actually raise if `sel` is a label but `coord` is not found. Otherwise we get confusing errors down the line, or potentially risk some unintentional conversions to `int`.

---

But isn't that already handled elsewhere? That should work the same for label based and positional indexing I guess.

---

I don't think so, because positional indexing works without coords. Give it a try with Scipp vs. ScippNexus (this branch), and see what exception types and error messages you get.

---

Hm, just noticed the positional indexing behavior is not the same for `scipp.DataGroup` and a scippnexus `File`.
In particular, this does not raise:
```python
with snx.File("60248-2022-02-28_2215.nxs") as f:
    raw = f["entry"][()]
    
raw['some_dimension_name',  :]
```
even if the dimension name does not exist.
But it raises in the example above.